src/layers/legacy/medVtkInria/vtkImageView/vtkImageView2D.cxx

```diff
@@ -1784,68 +1728,56 @@ int vtkImageView2D::GetFirstLayer() const
 /** Set/Get the input image to the viewer. */
 void vtkImageView2D::SetInput(vtkAlgorithmOutput* pi_poVtkAlgoOutput, vtkMatrix4x4 *matrix /*= 0*/, int layer /*= 0*/)
 {
-  vtkRenderer *renderer = 0;
-
-  if ( layer == 0 || IsFirstLayer(layer))
-  {
-      SetFirstLayer( pi_poVtkAlgoOutput, matrix, layer);
-  }
-  else // layer >0
-  {
-      this->AddLayer(layer);
-
-      if (!this->GetMedVtkImageInfo() || !this->GetMedVtkImageInfo()->initialized)
-      {
-          vtkErrorMacro (<< "Set input prior to adding layers");
-          return;
-      }
-
-      vtkAlgorithmOutput *reslicerOutputPort = this->ResliceImageToInput(pi_poVtkAlgoOutput, matrix);
-      if (!reslicerOutputPort)
-      {
-          vtkErrorMacro (<< "Could not reslice image to input");
-          return;
-      }
+    if (pi_poVtkAlgoOutput)
+    {
+        if (layer == 0 || IsFirstLayer(layer))
```

Comment:
i wonder if the "isFirstLayer" is really needed at the end. It has been mostly removed from `vtkImageView2D::SetFirstLayer` (the addLayer) since we never reach, and it is not in `vtkImageView3D::SetInput`

Reply:
Ok, I don't know if it's really useful